crates/sui-core/src/execution_cache/writeback_cache.rs

```diff
@@ -642,6 +637,11 @@ impl WritebackCache {
                 .get(tx)
                 .map(|o| o.clone())
             else {
+                // this case can only happen when a node is running with PassthroughCache, and
+                // then restarts with WritebackCache.
+                if self.is_tx_already_executed(tx).unwrap() {
```

Comment:
Since the cache selection now at epoch boundary, do we still need this? By having this, I wonder if it could mask some bug, since this is no longer expected.

Reply:
ah, good point. will remove.

Reply:
turns out we still need this, but for a different reason. comments updated